bentoml/_internal/resource.py

```diff
@@ -211,25 +214,29 @@ def query_os_cpu_count() -> int:
 #         raise ValueError(f"Invalid MEM resource limit '{mem}'")
 
 
-class NvidiaGpuResource(Resource[int], resource_id="nvidia.com/gpu"):
+class NvidiaGpuResource(Resource[t.List[int]], resource_id="nvidia.com/gpu"):
     @classmethod
-    def from_spec(cls, spec: t.Any) -> int:
-        if not isinstance(spec, (int, str)):
-            raise TypeError("NVidia GPU resource limit must be int or str")
-
-        if isinstance(spec, int):
-            return spec
+    def from_spec(cls, spec: t.Union[int, str, t.List[int | str]]) -> t.List[int]:
+        if not isinstance(spec, (int, str, t.List)):
+            raise TypeError(
+                "NVidia GPU resource limit must be int, str or a list specifing the exact GPUs to use."
```

Comment:
maybe "GPU device IDs".

Reply:
is this a bit too confusing?
if the user gives
`nvidia.com/gpu: 3`  then it is the number of resources
``nvidia.com/gpu: [3]` is the specific GPU number

Reply:
I can add more info in the doc section and this type error can point to that section of our doc